courses-rest/src/main/java/no/sikt/nva/fs/client/HttpUrlConnectionFsClient.java

```diff
@@ -0,0 +1,80 @@
+package no.sikt.nva.fs.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.util.Base64;
+import nva.commons.core.paths.UriWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HttpUrlConnectionFsClient implements FsClient {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HttpUrlConnectionFsClient.class);
+    public static final  String UNEXPECTED_STATUS_CODE_LOG_MESSAGE_PREFIX = "Unexpected response code from FS: ";
+    private static final String TEACHING_URI_PATH = "undervisning";
+    private static final String DB_IDENTIFIER_QUERY_PARAM_NAME = "dbId";
+    private static final String INSTITUTION_QUERY_PARAM_NAME = "emne.institusjon";
+    private static final String YEAR_QUERY_PARAM_NAME = "semester.ar";
+    private static final String LIMIT_QUERY_PARAM_NAME = "limit";
+    private static final String UNLIMITED = "0";
+    private final HttpClient httpClient;
+    private final ObjectMapper objectMapper;
+    private final String baseUri;
+    private final int institutionCode;
+    private final String username;
+    private final String password;
+
+    public HttpUrlConnectionFsClient(final ObjectMapper objectMapper, final String baseUri,
+                                     final int institutionCode, final String username,
+                                     final String password) {
+
+        this.httpClient = HttpClient.newBuilder().build();
+        this.objectMapper = objectMapper;
```

Comment:
the objectMapper is always JsonUtils.dtoObjectMapper. So we could either use this directly or (even better) create a staitic field that will be used everywhere